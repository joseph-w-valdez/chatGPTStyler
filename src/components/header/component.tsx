import React from "react";

// // // //

export function Header() {
    return (
        <div className="row">
            <div className="col-lg-12 text-center">
                <h1 className="text-lg">ChatGPT Styler</h1>
            </div>
        </div>
    );
}
