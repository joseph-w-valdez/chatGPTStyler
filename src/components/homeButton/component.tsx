import React from "react";
import css from "./styles.module.css";

interface HomeButtonProps {
    dataTestid: string;
    onClick: React.MouseEventHandler<HTMLButtonElement>;
    btnLabel: string;
}

export function HomeButton({
    dataTestid,
    onClick,
    btnLabel,
}: HomeButtonProps): JSX.Element {
    return (
        <button className={css.btn} data-testid={dataTestid} onClick={onClick}>
            {btnLabel}
        </button>
    );
}
