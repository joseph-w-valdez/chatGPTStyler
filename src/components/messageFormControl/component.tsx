import { SettingsType } from "@src/lib/utilities/googleStorage";
import React from "react";
import { useState, useEffect } from "react";

export interface MessageFormControlProps {
    settingsOptions: "message" | "text";
    setLiveChanges: React.Dispatch<React.SetStateAction<SettingsType>>;
    liveChanges: SettingsType;
    applyUpdates: (action: string, value: number | string) => void;
}

export function MessageFormControl({
    settingsOptions,
    setLiveChanges,
    liveChanges,
    applyUpdates,
}: MessageFormControlProps): JSX.Element {
    const [colorType, setColorType] = useState<string>("");
    const [inputMaxLength, setInputMaxLength] = useState<number>(30);
    const formatColor = (color: string) => {
        switch (colorType) {
            case "name":
                setInputMaxLength(30);
                return color;
            case "hex":
                setInputMaxLength(6);
                return `#${color}`;
            case "rgb":
                setInputMaxLength(11);
                return `rgb(${color})`;
            case "hsl":
                setInputMaxLength(9);
                return `hsl(${color})`;
            default:
                return color;
        }
    };

    const formatKey = (isUser: boolean): keyof SettingsType => {
        let formattedKey: keyof SettingsType;
        if (isUser) formattedKey = `${settingsOptions}ColorUserStyle`;
        else formattedKey = `${settingsOptions}ColorNonUserStyle`;
        return formattedKey;
    };
    // useEffect(() => {
    //     switch (colorType) {
    //         case "name":
    //             setPlaceholderValue("Color Name");
    //             setInputMaxLength(30);
    //             colorCode && setColorStyle(colorCode);
    //             break;
    //         case "hex":
    //             setPlaceholderValue("HEX");
    //             setInputMaxLength(6);
    //             colorCode.length === 6 && setColorStyle(`#${colorCode}`);
    //             break;
    //         case "rgb":
    //             setPlaceholderValue("R, G, B");
    //             setInputMaxLength(11);
    //             colorCode.length >= 5 && setColorStyle(`rgb(${colorCode})`);
    //             break;
    //         case "hsl":
    //             setPlaceholderValue("HUE, Saturation%, Light%");
    //             setInputMaxLength(9);
    //             colorCode.length >= 7 && setColorStyle(`hsl(${colorCode})`);
    //             break;
    //         default:
    //             setPlaceholderValue("Select Color Type");
    //             setColorType("");
    //             setColorCode("");
    //             setColorStyle("");
    //             setInputMaxLength(0);
    //     }
    //     colorLiveChange(colorStyle);
    // }, [colorType, colorCode, colorStyle]);

    return (
        <div className="flex flex-col justify-center items-center bg-indigo-400 rounded-md p-2 gap-1 font-medium">
            <label
                htmlFor={`${formatKey(true)}ColorType`}
                className=" text-white text-center"
            >
                {`Chat Message ${
                    formatKey(true) === "messageColorUserStyle"
                        ? "Background"
                        : "Text"
                } Color`}
            </label>
            <select
                id={`${formatKey(true)}`}
                className="p-1 rounded-md"
                onChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
                    setColorType(e.currentTarget.value)
                }
            >
                <option value="" disabled hidden>
                    Color Type
                </option>
                <option value="">Select</option>
                <option value="name">Name</option>
                <option value="hex">HEX</option>
                <option value="rgb">RGB</option>
                <option value="hsl">HSL</option>
            </select>
            <div className="grid grid-cols-2 gap-x-1">
                <input
                    className={`text-center w-full p-1 rounded-md ${
                        colorType
                            ? "animate-fade-in placeholder:text-black/30"
                            : "animate-fade-out placeholder:text-white/70 "
                    }`}
                    disabled={!colorType}
                    type="text"
                    id={`${formatKey(true)} TextColor`}
                    maxLength={inputMaxLength}
                    value={liveChanges[formatKey(true)]}
                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                        setLiveChanges({
                            ...liveChanges,
                            [formatKey(true)]: formatColor(
                                e.currentTarget.value,
                            ),
                        });
                        applyUpdates(
                            formatKey(true),
                            formatColor(e.currentTarget.value),
                        );
                    }}
                />
                <input
                    className={`text-center w-full p-1 rounded-md ${
                        colorType
                            ? "animate-fade-in placeholder:text-black/30"
                            : "animate-fade-out placeholder:text-white/70 "
                    }`}
                    disabled={!colorType}
                    type="text"
                    id={`${formatKey(false)} TextColor`}
                    maxLength={inputMaxLength}
                    value={liveChanges[formatKey(false)]}
                    onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                        setLiveChanges({
                            ...liveChanges,
                            [formatKey(false)]: formatColor(
                                e.currentTarget.value,
                            ),
                        });
                        applyUpdates(
                            formatKey(false),
                            formatColor(e.currentTarget.value),
                        );
                    }}
                />
                <span className="text-white rounded-md font-medium text-center p-0 m-0">
                    User
                </span>
                <span className="text-white rounded-md font-medium text-center p-0 m-0">
                    Chat-GPT
                </span>
            </div>
        </div>
    );
}
