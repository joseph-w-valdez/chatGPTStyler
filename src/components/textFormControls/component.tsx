import React from "react";
import { useState, useEffect } from "react";

export interface TextFormControlsProps {
    section: string;
    colorLiveChange: (colorStyle: string) => void;
    fontSizeOnChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
    fontWeightOnChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
}

export function TextFormControls({
    section,
    colorLiveChange,
    fontSizeOnChange,
    fontWeightOnChange,
}: TextFormControlsProps): JSX.Element {
    const [placeholderValue, setPlaceholderValue] =
        useState<string>("Select Color Type");
    const [colorType, setColorType] = useState<string>("");
    const [colorCode, setColorCode] = useState<string>("");
    const [colorStyle, setColorStyle] = useState<string>("");
    const [inputMaxLength, setInputMaxLength] = useState<number>();
 
    useEffect(() => {
        switch (colorType) {
            case "name":
                setPlaceholderValue("Color Name");
                setInputMaxLength(30);
                colorCode && setColorStyle(colorCode);
                break;
            case "hex":
                setPlaceholderValue("HEX");
                setInputMaxLength(6);
                colorCode.length === 6 && setColorStyle(`#${colorCode}`);
                break;
            case "rgb":
                setPlaceholderValue("R, G, B");
                setInputMaxLength(11);
                colorCode.length >= 5 && setColorStyle(`rgb(${colorCode})`);
                break;
            case "hsl":
                setPlaceholderValue("HUE, Saturation%, Light%");
                setInputMaxLength(9);
                colorCode.length >= 7 && setColorStyle(`hsl(${colorCode})`);
                break;
            default:
                setPlaceholderValue("Select Color Type");
                setColorType("");
                setColorCode("");
                setColorStyle("");
                setInputMaxLength(0);
        }
        colorLiveChange(colorStyle);
    }, [colorType, colorCode, colorStyle]);

    // // // //
    return (
        <div>
            <h1>{`${section} Text`}</h1>
            <hr />
            <label>
                Text Color:
                <select
                    id={`${section}ColorType`}
                    defaultValue={""}
                    onChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
                        setColorType(e.currentTarget.value)
                    }
                >
                    <option value="" disabled hidden>
                        Color Type
                    </option>
                    <option value="name">Name</option>
                    <option value="hex">HEX</option>
                    <option value="rgb">RGB</option>
                    <option value="hsl">HSL</option>
                </select>
                <input
                    disabled={colorType ? false : true}
                    type="text"
                    id={`${section}TextColor`}
                    placeholder={placeholderValue}
                    maxLength={inputMaxLength}
                    value={colorCode}
                    onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                        setColorCode(e.currentTarget.value)
                    }
                />
            </label>
            <br />
            <label>
                Font Size:
                <input
                    type="text"
                    id={`${section}FontSize`}
                    placeholder="pixel"
                    onChange={fontSizeOnChange}
                />
            </label>
            <br />
            <label>
                Font Weight:
                <select
                    id={`${section}FontWeight`}
                    defaultValue={""}
                    onChange={fontWeightOnChange}
                >
                    <option value="" disabled hidden>
                        Select a Font Weight
                    </option>
                    <option value="100">Thin</option>
                    <option value="200">Extra Light</option>
                    <option value="300">Light</option>
                    <option value="400">Normal</option>
                    <option value="500">Medium</option>
                    <option value="600">Semi Bold</option>
                    <option value="700">Bold</option>
                    <option value="800">Extra Bold</option>
                </select>
            </label>
        </div>
    );
}
