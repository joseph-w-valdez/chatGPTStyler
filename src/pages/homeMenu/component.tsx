import React, { MouseEventHandler } from "react";
import { HomeButton } from "@src/components/homeButton";
import { executeScript } from "@src/lib";

// // // //

const scrollToTopPosition = 0;
const scrollToBottomPosition = 9999999;

interface HomeMenuProps {
    textEditorRedirect: any;
    messageEditorRedirect: any;
}

// // // //

export function HomeMenu({
    textEditorRedirect,
    messageEditorRedirect,
}: HomeMenuProps): JSX.Element {
    return (
        <div className="grid gap-3 grid-cols-2 mt-3 w-full">
            <HomeButton
                dataTestid="scroll-to-top"
                onClick={() => {
                    executeScript(scrollToTopPosition);
                }}
                btnLabel="Scroll To Top"
            />
            <HomeButton
                dataTestid="scroll-to-botom"
                onClick={() => {
                    executeScript(scrollToBottomPosition);
                }}
                btnLabel="Scroll To Bottom"
            />
            <HomeButton
                dataTestid="text-editor"
                onClick={textEditorRedirect}
                btnLabel="Text Editor"
            />
            <HomeButton
                dataTestid="message-editor"
                onClick={messageEditorRedirect}
                btnLabel="Message Editor"
            />
        </div>
    );
}
