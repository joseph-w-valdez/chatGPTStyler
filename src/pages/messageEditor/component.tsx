import React, { useState } from "react";
import { SettingsType } from "@src/lib/utilities/googleStorage";
import { MessageFormControl } from "@src/components/messageFormControl";
import { FormButtons } from "@src/components/formButtons/component";
import { MessageSliderControls } from "./components/MessageSliderControls";

export interface MessageEditorProps {
    settings: SettingsType;
    setSettings: React.Dispatch<React.SetStateAction<SettingsType>>;
}

export function MessageEditor({
    settings,
    setSettings,
}: MessageEditorProps): JSX.Element {
    const [isEditing, setIsEditing] = useState(false);
    const [liveChanges, setLiveChanges] = useState<SettingsType>({
        ...settings,
    });
    const sendMessageToRuntime = (
        action: keyof SettingsType | "restoreSettings",
        value?: number | string | SettingsType,
    ) => {
        setIsEditing(true);
        chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
            chrome.tabs.sendMessage(tabs[0].id || 0, {
                action: action,
                arg: value,
            });
        });
    };

    return (
        <div
            className={`grid grid-cols-1 gap-y-3 px-3 pb-2 ${
                !isEditing ? "animate-fade-in" : ""
            }`}
        >
            <MessageSliderControls
                setLiveChanges={setLiveChanges}
                liveChanges={liveChanges}
                sendMessageToRuntime={sendMessageToRuntime}
            />
            <MessageFormControl
                setLiveChanges={setLiveChanges}
                liveChanges={liveChanges}
                sendMessageToRuntime={sendMessageToRuntime}
            />
            <FormButtons
                isEditing={isEditing}
                setIsEditing={setIsEditing}
                liveChanges={liveChanges}
                setLiveChanges={setLiveChanges}
                settings={settings}
                setSettings={setSettings}
                sendMessageToRuntime={sendMessageToRuntime}
            />
        </div>
    );
}
