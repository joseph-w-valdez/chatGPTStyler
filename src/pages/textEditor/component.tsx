import React, { useEffect } from "react";
import { useState } from "react";
import { TextFormControls } from "@src/components/textFormControls";
import {
    SettingsType,
    saveOptionsToStorage,
} from "@src/lib/utilities/googleStorage";
import css from "./styles.module.css";

interface TextEditorProps {
    settings: SettingsType;
    setOptions: React.Dispatch<React.SetStateAction<SettingsType>>;
}

export function TextEditor({
    settings,
    setOptions,
}: TextEditorProps): JSX.Element {
    const [isEditing, setIsEditing] = useState(false);
    const [liveChanges, setLiveChanges] = useState<SettingsType>(settings);

    // useEffect(() => {
    //     getOptionsFromStorage((savedOptions) => {
    //         setOptions(savedOptions);
    //         setLiveSettings(savedOptions);
    //         console.log(liveSettings);
    //     });
    // }, []);

    const sendMessageToRuntime = (action: string, value: number | string) => {
        setIsEditing(true);
        console.log("action", action);
        console.log("value", value);
        chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
            chrome.tabs.sendMessage(tabs[0].id || 0, {
                action: action,
                arg: value,
            });
        });
    };
    return (
        <div className="grid grid-cols-1 gap-4">
            <TextFormControls
                settingsOptions={"Size"}
                liveChanges={liveChanges}
                setLiveChanges={setLiveChanges}
                applyUpdates={sendMessageToRuntime}
            />
            <TextFormControls
                settingsOptions={"Weight"}
                liveChanges={liveChanges}
                setLiveChanges={setLiveChanges}
                applyUpdates={sendMessageToRuntime}
            />
            <div className="grid grid-cols-4 gap-1">
                <button
                    className={`${css.btn} col-span-2`}
                    onClick={() => {
                        setLiveChanges({
                            messageMaxWidthStyle: "95",
                            messageColorUserStyle: "",
                            messageColorNonUserStyle: "",
                            messagePaddingStyle: "10",
                            messageBorderRadiusStyle: "5",
                            inputBoxMaxWidthStyle: "70",
                            textColorUserStyle: "",
                            textColorNonUserStyle: "",
                            textSizeUserStyle: "",
                            textSizeNonUserStyle: "",
                            textWeightUserStyle: "",
                            textWeightNonUserStyle: "",
                        });
                        sendMessageToRuntime("restoreDefaultSettings", "");
                        setIsEditing(true);
                    }}
                >
                    Restore Default
                </button>
                <button
                    disabled={!isEditing}
                    className={`${css.btn}`}
                    onClick={() => {
                        saveOptionsToStorage(liveChanges);
                        setLiveChanges(liveChanges);
                        setIsEditing(false);
                    }}
                >
                    Save
                </button>
                <button
                    disabled={!isEditing}
                    className={`${css.btnRed}`}
                    onClick={() => {
                        setOptions({ ...liveChanges });
                        sendMessageToRuntime("restoreUserSettings", "");
                        setIsEditing(false);
                    }}
                >
                    Cancel
                </button>
            </div>
        </div>
    );
}
