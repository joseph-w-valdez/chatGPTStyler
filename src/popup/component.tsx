import React from "react";
import { useState, useEffect } from "react";
import browser from "webextension-polyfill";
import css from "./styles.module.css";
import { Header } from "@src/components/header/Header";
import {
    SettingsType,
    getOptionsFromStorage,
    saveOptionsToStorage,
} from "@src/lib/utilities/googleStorage";
import { MessageEditor } from "@src/pages/messageEditor";
import { HomeMenu } from "@src/pages/homeMenu";
import { defaultSettings } from "@src/shared/utils/data";

export function Popup(): JSX.Element {
    const [settings, setSettings] = useState<SettingsType>({
        ...defaultSettings,
    });

    const [page, setPage] = useState<string>("Home Menu");
    // Sends the `popupMounted` event
    useEffect(() => {
        browser.runtime.sendMessage({ popupMounted: true });
    }, []);

    // Load options from storage when the popup is opened
    useEffect(() => {
        getOptionsFromStorage((savedOptions) => {
            setSettings(savedOptions);
            console.log("loaded options from storage", savedOptions);
        });
    }, []);

    // Renders the component tree
    return (
        <div className={css.popupContainer}>
            <div className="w-full">
                <Header page={page} setPage={setPage} />
                <hr className="mb-2" />
                {page === "Message Editor" ? (
                    <MessageEditor
                        settings={settings}
                        setSettings={setSettings}
                    />
                ) : (
                    <HomeMenu setPage={setPage} />
                )}
            </div>
        </div>
    );
}
