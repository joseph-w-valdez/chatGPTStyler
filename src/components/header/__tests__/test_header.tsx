import * as React from "react";
import { Header } from "../component";
import renderer from "react-test-renderer";

it("component renders", () => {
    const tree = renderer
        .create(<Header page="string" setPage={() => {}} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
