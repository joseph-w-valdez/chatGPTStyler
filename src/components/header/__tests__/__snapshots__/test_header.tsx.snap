// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<div
  className="p-3 text-center relative"
>
  <h1
    className="text-lg"
  >
    ChatGPT Styler
  </h1>
  <button
    className="btn absolute inset-y-0 right-0 animate-fade-in m-2"
    onClick={[Function]}
  >
    Back
  </button>
</div>
`;
